src/psyclone/psyir/nodes/loop.py

```diff
@@ -111,9 +113,42 @@ def __init__(self, parent=None, variable_name="", valid_loop_types=None,
         # TODO replace iterates_over with iteration_space
         self._iterates_over = "unknown"
 
-        self._variable_name = variable_name
+        self._check_variable(variable)
+        self._variable = variable
         self._id = ""
 
+    @staticmethod
+    def _check_variable(variable):
+        '''The loop variable should be a scalar integer. Check that this is
+        the case and raise an exception if not.
+
+        :param variable: the loop iterator.
+        :type variable: :py:class:`psyclone.psyir.symbols.DataSymbol`
```

Comment:
or NoneType, and adjust the method description docstring

Reply:
Actually I've changed the code so NoneType is no longer allowed. See my response to your next comment.